src/SmartComponents/Actions/ListActions.js

```diff
@@ -39,10 +41,12 @@ class ListActions extends Component {
         this.setPage = this.setPage.bind(this);
         this.setPerPage = this.setPerPage.bind(this);
         this.parseProductCode = this.parseProductCode.bind(this);
+        this.onSearch = this.onSearch.bind(this);
```

Comment:
outta curiosity, why are we doing this here (and everywhere else)?

Reply:
ah got it, when a function relies on the component state (or external inputs), we gotta bind it

Reply:
the binding is necessary to make `this` work in the callback